src/users/views.py

```diff
@@ -249,18 +249,21 @@ def filter_queryset(self, queryset):
             query = self.request.GET['query']
             query = query.split(' ')
 
-            
             for item in query:
                 item_alias = trans_dict.get(item.lower(),item)
                 queryset = queryset.filter(
                     Q(major__name__icontains = item) | 
                     Q(profile__year__icontains = item) | 
                     Q(profile__user__first_name__icontains = item) |
-                    Q(profile__user__last_name__icontains = item)|
+                    Q(profile__user__last_name__icontains = item) |
+                    Q(minor__name__icontains = item) |
+                    Q(courses__name__icontains = item) |
                     Q(major__name__icontains = item_alias) | 
                     Q(profile__year__icontains = item_alias) | 
                     Q(profile__user__first_name__icontains = item_alias) |
-                    Q(profile__user__last_name__icontains = item_alias)
+                    Q(profile__user__last_name__icontains = item_alias) |
+                    Q(minor__name__icontains = item_alias) |
+                    Q(courses__name__icontains = item_alias)
```

Comment:
this long list is getting silly.  I tried to find a nice way of doing it, but there's no built in always false Q object.  Here's my attempt though
```python
filter_keywords = [
    'major__name__icontains', 
    'profile__year__icontains', 
    ...
]
query = Q(pk__in=[]) # always false query: https://stackoverflow.com/questions/35893867/always-false-q-object

for keyword in filter_keywords:
    filter_kwarg = {keyword: item_alias}
    query |= Q(**filter_kwarg)

queryset = queryset.filter(query)
```


Up to you if you think it's prettier

Reply:
On second thought, that does not look prettier.  The cleanest way would be to use a reducer, but functional programming is hard to approach.